src/test/rpc/JSONRPC_test.cpp

```diff
@@ -48,18 +49,26 @@ struct TxnTestData
     // The expMsg array contains the expected error string for the above cases.
     char const* const expMsg[4];
 
-    // Default and copy ctors should be deleted, but that displeases gcc 4.6.3.
-//  TxnTestData () = delete;
-//  TxnTestData (TxnTestData const&) = delete;
-//  TxnTestData (TxnTestData&&) = delete;
+    constexpr TxnTestData (char const* description_, int line_,
+        char const* json_, std::initializer_list<char const*> expMsg_)
+    : description (description_)
+    , line (line_)
+    , json (json_)
+    , expMsg {expMsg_.begin()[0],
+        expMsg_.begin()[1], expMsg_.begin()[2], expMsg_.begin()[3]}
```

Comment:
We needed to use a `std::initializer_list` here instead of a `std::array` in order to make this `constexpr`. I think the code is clearer with the `std::array` (we know the size should be 4 elements and `expMst_[0]` is nicer than `expMst_.begin()[0]`). I'd choose to use `std::array` instead of `initializer_list` here at the cost of losing the `constexpr`. For code that runs once in a unit test, any extra speedup by using a constexpr wouldn't be noticed anyway.

Reply:
I'd probally also make the members `std::string` instead of `char const*` so I'd never have to think about lifetime issues. I doubt the extra allocs to do that would be noticed. However, it's fine as-is as well.

Reply:
Actually I didn't realize `array` would work here at all.  Nice!

That being said, if we can use `constexpr` here and that costs the use of `initializer_list`, that sounds like a win to me.  But maybe add a comment that it can be changed to `array` when we move to C++17?

Reply:
News flash:  `const array` should allow `constexpr` to work in C++14! :-)

Reply:
Great catch, you two.  The test was originally written for C++11, where `std::array` was not `constexpr` enough to replace a C-style array.  Now that we're using C++14, the switch from C-style array to `std::array` worked like a charm.  Thanks for spotting that.